src/TextArea.tsx

```diff
@@ -0,0 +1,164 @@
+import styled, {css} from 'styled-components'
+import React, {TextareaHTMLAttributes, ReactElement} from 'react'
+import {FormValidationStatus} from './utils/types/FormValidationStatus'
+import sx, {SxProp} from './sx'
+import {get} from './constants'
+
+export type TextAreaProps = {
+  /**
+   * Apply inactive visual appearance to the TextArea
+   */
+  disabled?: boolean
+  /**
+   * Indicates whether the TextArea must be checked
+   */
+  required?: boolean
+  /**
+   * Indicates whether the TextArea validation state
+   */
+  validationStatus?: FormValidationStatus
+  /**
+   * Block
+   */
+  block?: boolean
+  /**
+   * Allows resizing of the textarea
+   */
+  resize?: 'none' | 'both' | 'horizontal' | 'vertical'
+  /**
+   * Applies a custom width to the textarea
+   */
+  width?: number
+  /**
+   * Applies a custom height to the textarea
+   */
+  height?: number
```

Comment:
Instead of `height`, maybe we should stick closer to the HTML API and use `rows?: number`.

Reply:
We could take it a step further and say that the textarea would render with the specified number of `rows`, but if the user input goes beyond that, the height would adjust to fit the text.

If we wanted to prevent a textarea from growing too high, we could have a `maxRows` prop that limits the height of the textarea. After the user inputs a number of lines that is greater than `maxRows`, it would just scroll vertically.

Reply:
+1 for `rows/cols` (which is natively supported by textarea). If a consumer wants to explicitly set the `width/height`, they can use the `sx` prop

Reply:
So rows/cols are still supported natively and for most users would be perfectly fine. It's definitely a bit confusing here because it’s not explicitly listed in the prop table, so it might appear like I was recommending one over the other. I don't agree that they are equivalent to explicit height and width however, which provide more granular control over appearance. 

Why not just use `sx` for this? Maybe 🤷 . I feel like we over-rely on `sx` too much already to handle common use-cases, which is a sentiment I’ve heard on a few occasions by users. So height and width were really just there for improved ergonomics and convenience.

Reply:
I still prefer `rows` over `height` because it sets the height to fit a specific number of lines.

I agree that `cols` feels weird, and I'd be happy with using `sx` to set width. I assume `cols` was added to support vertical writing modes.

Reply:
height has been removed 👍